backends/ubpf/tests/environment/setup-switch.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Orange
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Dependencies
+sudo apt-get -y install libnuma-dev clang-6.0 libc6-dev-i386
+sudo apt-get -y install flex bison
+sudo python -m pip install scapy
+
+# P4c dependencies
+sudo apt-get -y install cmake g++ git automake libtool libgc-dev bison flex libfl-dev libgmp-dev libboost-dev libboost-iostreams-dev libboost-graph-dev llvm pkg-config python python-scapy python-ipaddr python-ply tcpdump
+
+# Protobuf
+cd /vagrant
+if [ ! -d "/vagrant/protobuf" ]
+then
+  git clone https://github.com/google/protobuf.git
+  cd protobuf
+  git checkout v3.2.0
+  export CFLAGS="-Os"
+  export CXXFLAGS="-Os"
+  export LDFLAGS="-Wl,-s"
+  ./autogen.sh
+  ./configure
+  make
+  sudo make install
+  sudo ldconfig
+  unset CFLAGS CXXFLAGS LDFLAGS
+  cd python
+  sudo python setup.py install
+  cd ../..
+fi
+
+# Install clang-10
+if ! type "/vagrant/llvm-project/build/bin/clang" > /dev/null; then
+  cd /vagrant
+  git clone https://github.com/P4-Research/llvm-project
```

Comment:
so you have your own clone of llvm as well?
it would be nice to use an off-the-shelf version

Reply:
Yep, we are aware of that.

This is related to the limitation of BPF itself. It was also discussed in regards of p4c-xdp [1]. As we are in the userspace, we can simply increase the BPF's stack size. However, it requires to have a new flag in `clang` compiler (this is also on our TODO list), as proposed in this thread. 

Currently, we have our fork of `clang` / llvm, in which the stack size is increased and hardcoded.

[1] https://github.com/vmware/p4c-xdp/issues/22
[2] http://lists.llvm.org/pipermail/cfe-users/2017-March/001126.html